### PR TITLE
feat(Language): Redesign of Language settings screen

### LINKED ISSUE #4917: Language & Currency settings screen
needs https://github.com/status-im/StatusQ/issues/563

### Description

https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?node-id=701%3A74776

Screen Shot 2022-02-26 at 1 13 18 PM

* [x] Rename setting section "Language" to  "Language & Currency". -- Subtask [5384](https://github.com/status-im/status-desktop/issues/5384)

* [x] Replace Language picker with the Select Component -- Subtask [5384](https://github.com/status-im/status-desktop/issues/5384)

Screen Shot 2022-02-26 at 1 13 39 PM

Screen Shot 2022-02-26 at 1 13 53 PM

* [x] Add Display Currency Picker with the Select Component -- Subtask [5385](https://github.com/status-im/status-desktop/issues/5385)

* [x] Make wallet and any currency related UI code use chosen Display Currency -- Subtask [5385](https://github.com/status-im/status-desktop/issues/5385)

* [x] Add Date & Time Format and make relevant UI code use the chosen Format -- Subtask [5386](https://github.com/status-im/status-desktop/issues/5386)

### LINKED ISSUE #5384: Language settings screen
Subtask of [#4917](https://github.com/status-im/status-desktop/issues/4917)

Needs https://github.com/status-im/StatusQ/issues/563

### Description

https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?node-id=701%3A74776

* [x] Rename setting section "Language" to  "Language & Currency"

* [x] Replace Language picker with the Select Component

### PR DESCRIPTION
**Pull request by noeliaSD**
Subtask of #4917. Closes #5384 

### What does the PR do

Renamed setting section "Language" to "Language & Currency".

Updated link to StatusQ - Needed new `StatusListPicker` component.

Added language model in `LanguageStore` and created first view approach for `Language` setting.

Added temporal confirmation dialog for **Linux** OS (since it is currently needed to restart the app to apply changes). It should be addressed in [task #5387](https://github.com/status-im/status-desktop/issues/5387).

Removed no longer used files.

### Affected areas

Settings / Language

### Screenshot of functionality

https://user-images.githubusercontent.com/97019400/161787793-7afc66aa-de35-40ee-9f29-5db9ddcfb771.mov

### CODE REVIEW

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (stefandunca) [inline ui/app/AppLayouts/Profile/views/LanguageView.qml:76]: How long is this pause on your machine? Will it grow with an entry-level laptop? How about in the future when the complexity grows? Won't a popup with a message like "Loading the new translation" provide a better future-proof user experience?

**Review by stefandunca — APPROVED** [review]
Great work. Just some questions/curiosities from my side.

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Profile/views/LanguageView.qml:23]: can we rename this to something more meaningful like _internal or something?

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Profile/views/LanguageView.qml:42]: is the z order really needed in this case?

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Profile/views/LanguageView.qml:43]: add braces 560 - (2 * Style.current.padding)

**Review by Khushboo-dev-cpp — COMMENTED** [review]
Looks good. :) only some small comments

**Comment by abhi-status** [comment]
tested -> https://ethstatus.testrail.net/index.php?/runs/view/8911&group_by=cases:section_id&group_order=asc

**Review by abhi-status — APPROVED** [review]

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/views/LanguageView.qml:76]: We can ask @John-44 about the design / loading icon / message. 

Now what it's happening without the pause is that when user selects a new language it takes less than 1 second to execute the action so the UI seems to respond slow by marking the radiobutton this 1 second later. With the pause incorporated the UI is inmediately changed, so no visual block, but the action itself is executed 100 ms later (no perception to the user).

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/views/LanguageView.qml:23]: I would like to keep as it is because of [QML Convention from Qt - Private properties](https://doc.qt.io/archives/qt-4.8/qml-coding-conventions.html) and discussion [here](https://github.com/status-im/StatusQ/pull/594#discussion_r832494124). WDYT?

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/views/LanguageView.qml:42]: Yes. Since we are using `StatusListPicker` that contains a **drop-down list**, it is needed to have the drop-down list on top.

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/views/LanguageView.qml]: Sure! Done!

File: ui/app/AppLayouts/Profile/views/LanguageView.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/views/LanguageView.qml:76]: Ok! Let's do like this. Here the [#task](https://github.com/status-im/status-desktop/issues/5400) to decide how to proceed!

**Comment by noeliaSD** [comment]
Added `do not merge` tag because it is needed to update the `StatusQ` link but still waiting for approvals on the related `StatusQ` PRs.

**Review by iurimatias — APPROVED** [review]
tested https://ethstatus.testrail.net/index.php?/runs/view/8924

**Pull request merged**
